pacehrh/R/pace_stochasticity_config.R

```diff
@@ -59,6 +59,54 @@ loadChangeRateLimits <- function(changeRateLimitsSheetName = .defaultChangeRateL
                                .changeRateLimitsMetaData,
                                convertType = FALSE)
 
+  limitData <- .validateChangeRateLimits(limitData)
+
+  return(limitData)
+}
+
+.validateChangeRateLimits <- function(limitData) {
+  if (is.null(limitData)) {
+    return(NULL)
+  }
+
+  # Sanity check data types (should already have been done as part of
+  # validateTableAgainstSchema())
+  types <- sapply(limitData, typeof)
+  names(types) <- NULL
+
+  if (!isTRUE(all.equal(types, .changeRateLimitsColumnTypes))) {
+    return(NULL)
+  }
+
+  .clearRow <- function(rowNum) {
+    limitData[rowNum, "Min"] <<- NA_real_
+    limitData[rowNum, "Max"] <<- NA_real_
+  }
+
+  .isValidRow <- function(min, max) {
+    if (is.na(min) | is.na(max)) {
+      return(FALSE)
+    }
+
+    # Negative values aren't allowed
+    if ((min < 0.0) | (max < 0.0)) {
+      return(FALSE)
+    }
+
+    # The maximum limit must be greater than or equal to the minimum
+    if (min > max) {
+      return(FALSE)
+    }
+
+    return(TRUE)
+  }
+
+  for (i in 1:NROW(limitData)) {
+    if (!.isValidRow(limitData[i, "Min"], limitData[i, "Max"])) {
+      .clearRow(i)
```

Comment:
can you add some message before clearing the invalid data so user can be warned?

Reply:
Sure, but can we do that in a separate follow-up PR?

Reply:
Alternatively, I could add the expected behavior for these scenarios when documentation is updated (issue #246). "when the user enters limits considered invalid by the model (i.e. negative values for the change rate limits, or a min value greater than a max value), the limits will be reset to empty and the model would ran as if no limits on change rates."

Reply:
Great idea Rui. I think we should do that anyway!